setup_payram_agents.sh

```
@@ -2240,17 +2240,44 @@ cmd_run() {
+		# digit + special). State the rule UP FRONT, validate locally with a
+		# retry loop, and offer auto-generation - never let a human discover
+		# the rule via an opaque INVALID_JSON_DATA 400.
 		local password="${PAYRAM_PASSWORD:-}"
-		[[ -z "$password" ]] && read -s -p "Password: " password && echo
+		while ! password_meets_complexity "$password"; do
+			if [[ -n "$password" ]]; then
+				echo "That password doesn't meet the rule - try again."
+			fi
+			echo "Password rule: at least 8 characters, with a lowercase, an uppercase,"
+			echo "a digit, and a special character."
+			read -s -p "Password (press Enter to auto-generate a compliant one): " password && echo
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)